### PR TITLE
WIP: Amino & proto3 tests/experiments

### LINKED ISSUE #166: proto compatibility issues
Collecting some issues (or sometimes just differences) I stumbled upon while testing here:

- floating types not supported in amino (`panic: floating point types are unsafe for go-amino`) but they are supported in protobuf (I know this was discussed somewhere but I can't find it right now); this is a bit annoying because it prevents just using their generated test structs (of course this annoyance no reason to support them though)
- something seems different regarding (u)int32 encoding ([example](https://gist.github.com/Liamsi/6ddd3e824c351799905f0c0a7950aecf)), ~~could just be the field numbering, though~~(this is definitely different) See another [example](https://gist.github.com/Liamsi/efe2c0816f686d1c631cd64dec1e3aa2), separate issue dealing with fixed size types: https://github.com/tendermint/go-amino/issues/168
- to be continued ...

simple struct 

related #163 #164

### PR DESCRIPTION
**Pull request by liamsi**
resolves some of the issues tracked here:
resolves #166

### CODE REVIEW

**Comment by jaekwon** [comment]
Pushed to develop all the changes including the BigEndian LittleEndian fix.
More comments on the file diff view on this PR... the tests still need to be ported over in a new PR or rebase.
https://github.com/tendermint/go-amino/commit/f297b23166b16ad60976b6ba11a14587bd9b479b

File: binary-encode.go [hunk 1]
```diff
@@ -84,12 +84,14 @@ func (cdc *Codec) encodeReflectBinary(w io.Writer, info *TypeInfo, rv reflect.Va
 
 	case reflect.Int64:
 		if fopts.BinVarint {
-			err = EncodeVarint(w, rv.Int())
+			err = EncodeVarint(w, uint64(rv.Int()))
```
Comment (jaekwon) [inline binary-encode.go:87]: We can keep the code as is, EncodeVarint() is the encoding for Proto3 signed varints (field type is sint64 and sint32).

File: encoder.go [hunk 2]
```diff
@@ -11,31 +11,113 @@ import (
...
+	binary.LittleEndian.PutUint64(buf[:], uint64(i))
 	_, err = w.Write(buf[:])
 	return err
 }
 
-func EncodeVarint(w io.Writer, i int64) (err error) {
-	var buf [10]byte
-	n := binary.PutVarint(buf[:], i)
-	_, err = w.Write(buf[0:n])
+func EncodeVarint(w io.Writer, v uint64) (err error) {
```
Comment (jaekwon) [inline encoder.go:35]: This is the encoding function for Proto3 unsigned varints.  Signed varints should use the previous logic, binary.PutVarint()...

**Comment by liamsi** [comment]
continued in #176

**Pull request closed without merging**